flocker/testtools/_flaky.py

```diff
@@ -22,10 +37,179 @@ def flaky(jira_key):
 
     # TODO (see FLOC-3281):
     # - allow specifying which exceptions are expected
-    # - retry on expected exceptions
-    #   - parametrize on retry options
     # - provide interesting & parseable logs for flaky tests
 
+    if isinstance(jira_keys, basestring):
+        jira_keys = [jira_keys]
+
+    annotation = _FlakyAnnotation(
+        jira_keys=pset(jira_keys), max_runs=max_runs, min_passes=min_passes)
+
     def wrapper(test_method):
+        setattr(test_method, _FLAKY_ATTRIBUTE, annotation)
         return test_method
+
     return wrapper
+
+
+def _get_flaky_annotation(case):
+    """
+    Get the flaky decoration detail from ``case``.
+
+    :param TestCase case: A test case that might have been decorated with
+        @flaky.
+    :return: ``None`` if not flaky, or a ``pmap`` of the flaky test details.
+    """
+    # XXX: Is there a public way of doing this?
```

Comment:
:grey_question: Looks like you are doing it in the standard way, or at least the way it's done within testtools:
https://github.com/testing-cabal/testtools/blob/master/testtools/testcase.py#L216

Might I suggest making this TODO a little more future-actionable by openning an issue against testtools to make this method public, and referencing that issue here?